### PR TITLE
Leverage basic AWSPlacementGroup for AWS PG membership

### PR DESCRIPTION
**Pull request by JoelSpeed**
This is a very basic implementation that requires users to manually create a PG on AWS and then an `AWSPlacementGroup` resource within the cluster.

Eventually, a controller will do the creation of the placement group for them, but we are going to make this manual in the first iteration.

To use this:

On AWS, create a placement group and remember the name, in this example the strategy is set to `cluster` on AWS

On OpenShift, add a resource like below:
```
apiVersion: machine.openshift.io/v1
kind: AWSPlacementGroup
metadata:
  name: <name-of-placement-group>
  namespace: openshift-machine-api
spec:
  credentialsSecret:
    name: aws-cloud-credentials
  managementSpec:
    managementState: Unmanaged
```
Add the name to the placement group to the Machine provder spec
```
placement:
  group:
    name: <name-of-placement-group>
```

Eventually this will be extended to check that the `AWSPlacementGroup` is ready before leveraging the placement group

### CODE REVIEW

File: pkg/actuators/machine/instances.go [hunk 1]
```diff
@@ -525,3 +506,50 @@ func getInstanceMarketOptionsRequest(providerConfig *machinev1.AWSMachineProvide
 
 	return instanceMarketOptionsRequest
 }
+
+// constructInstancePlacement configures the placement options for the RunInstances request
+func constructInstancePlacement(machine *machinev1.Machine, machineProviderConfig *machinev1.AWSMachineProviderConfig, client runtimeclient.Client) (*ec2.Placement, error) {
+	placement := &ec2.Placement{}
+	if machineProviderConfig.Placement.AvailabilityZone != "" && machineProviderConfig.Subnet.ID == nil {
```
Comment (elmiko) [inline pkg/actuators/machine/instances.go:513]: what happens if the user specifies an AvailabilityZone and a Subnet.ID ?

File: pkg/actuators/machine/instances_test.go [hunk 2]
```diff
@@ -965,6 +968,114 @@ func TestLaunchInstance(t *testing.T) {
...
+						},
+					},
+				},
+				Placement: &ec2.Placement{
+					GroupName:       aws.String(stubPlacementGroupName),
+					PartitionNumber: aws.Int64(1),
+				},
+				UserData: aws.String(""),
+			},
+		},
```
Comment (elmiko) [inline pkg/actuators/machine/instances_test.go:1078]: should there be a test to exercise placement with an availability zone?

**Review by elmiko — COMMENTED** [review]
this mostly makes sense to me Joel, just a few questions

File: pkg/actuators/machine/actuator_test.go [hunk 3]
```diff
@@ -10,7 +10,8 @@ import (
 	"github.com/golang/mock/gomock"
 	. "github.com/onsi/gomega"
 	configv1 "github.com/openshift/api/config/v1"
-	machinev1 "github.com/openshift/api/machine/v1beta1"
+  machinev1 "github.com/openshift/api/machine/v1"
```
Comment (damdo) [inline pkg/actuators/machine/actuator_test.go:13]: Indentation :)

File: pkg/actuators/machine/actuator_test.go [hunk 4]
```diff
@@ -24,7 +25,8 @@ import (
 
 func init() {
 	// Add types to scheme
-	machinev1.AddToScheme(scheme.Scheme)
+	machinev1beta1.AddToScheme(scheme.Scheme)
+  machinev1.Install(scheme.Scheme)
```
Comment (damdo) [inline pkg/actuators/machine/actuator_test.go:29]: Also here

**Review by damdo — COMMENTED** [review]
Went through the code and tested the feature manually on a testing cluster. 
Aside from some indentation to fix, it LGTM.

**Review by damdo — COMMENTED** [review]
/lgtm

**Review by alexander-demichev — APPROVED** [review]
/approve

**Pull request merged**